http_server/string_seq_match_test.py

```diff
@@ -1,85 +1,86 @@
 import random
-import sys
 from amaranth.sim import Simulator
 
 from string_match import StringMatch
 from string_alt_match import StringAltMatch
 from string_seq_match import StringSeqMatch
 
-dut = StringSeqMatch(sequence=[
-    StringAltMatch(alternatives=[
-        StringMatch("GET", match_case=False),
-        StringMatch("POST", match_case=False),
-    ]),
-    StringMatch(" "),
-    StringAltMatch(alternatives=[
-        StringMatch("/"),
-        StringMatch("/style.css"),
-        StringMatch("/index.html"),
-    ]),
-    StringMatch(" HTTP/1"),
-    StringAltMatch(alternatives=[
-        StringMatch(".0\r\n"),
-        StringMatch("\r\n"),
-    ]),
-])
-
-
-async def run_sequence(ctx, input: str):
-    """
-    Runs the input sequence into the DUT until accepted, rejected,
-    or indeterminate after the input has been passsed.
-
-    Returns which match if the input was accepted;
-    otherwise returns None.
-    """
-    ctx.set(dut.reset, 1)
-    await ctx.tick()
-    ctx.set(dut.reset, 0)
-    ctx.set(dut.input.valid, 0)
-
-    assert ctx.get(dut.accepted) == 0
-    assert ctx.get(dut.rejected) == 0
-
-    count = 0
-    input_len = len(input)
-    while True:
-        if ctx.get(dut.accepted) or ctx.get(dut.rejected):
-            break
-        if count == input_len:
-            break
-        assert ctx.get(dut.input.ready)
-        valid = random.randint(0, 1)
-        ctx.set(dut.input.valid, valid)
-        if valid:
-            # Advance the input on this cycle.
-            ctx.set(dut.input.payload, ord(input[0]))
-            input = input[1:]
-            count += 1
+
+def run_test_sequence(input: str):
+
+    dut = StringSeqMatch(sequence=[
+        StringAltMatch(alternatives=[
+            StringMatch("GET", match_case=False),
+            StringMatch("POST", match_case=False),
+        ]),
+        StringMatch(" "),
+        StringAltMatch(alternatives=[
+            StringMatch("/"),
+            StringMatch("/style.css"),
+            StringMatch("/index.html"),
+        ]),
+        StringMatch(" HTTP/1"),
+        StringAltMatch(alternatives=[
+            StringMatch(".0\r\n"),
+            StringMatch("\r\n"),
+        ]),
+    ])
```

Comment:
Hrm, I guess this is probably how I should have used existing structures instead for #27 ... I guess disadvantage for this one is that the path will always only match "/" (if I'm understanding string_alt_match.py correctly), but that probably could get fixed?

Reply:
Right, yeah, `StringAltMatch` doesn't handle prefix/suffix alternatives correctly -- it's greedy.

I think the more bespoke start-line parser is fine, I'm often bit by trying to generalize too much too soon :)